python/lsst/ts/tunablelaser/mock_server.py

```diff
@@ -113,21 +114,39 @@ def __init__(self, host=tcpip.LOCAL_HOST, port=0) -> None:
         self.device = MockNP5450()
         self.log = logging.getLogger(__name__)
         self.read_loop_task = asyncio.Future()
-        super().__init__(
-            name="TempCtrl Mock Server",
-            host=host,
-            port=port,
-            log=self.log,
-            terminator=b"\r",
-            encoding="ascii",
-        )
+        try:
+            ip_address(host)
+            super().__init__(
+                name="TempCtrl Mock Server",
+                host=host,
+                port=port,
+                log=self.log,
+                terminator=b"\r",
+                encoding="ascii",
+            )
+        except ValueError:
+            self.log.error(
+                f"TempCtrlServer hostname was invalid, assuming temp ctrler unconnected: {host}"
+            )
+            super().__init__(
+                name="TempCtrl Mock Server",
+                host=tcpip.LOCAL_HOST,
+                port=50,
```

Comment:
Don't default to 50 as that is a privileged port and can cause issues with the CI.